src/test/kotlin/uk/gov/dluhc/printapi/jobs/FinalRetentionPeriodDataRemovalJobIntegrationTest.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.dluhc.printapi.jobs
+
+import ch.qos.logback.classic.Level
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+import uk.gov.dluhc.printapi.config.IntegrationTest
+import uk.gov.dluhc.printapi.database.entity.PrintRequest
+import uk.gov.dluhc.printapi.testsupport.TestLogAppender
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildCertificate
+import java.time.LocalDate
+import java.util.UUID
+
+internal class FinalRetentionPeriodDataRemovalJobIntegrationTest : IntegrationTest() {
+
+    @Autowired
+    private lateinit var testPrintRequestRepository: TestPrintRequestRepository
```

Comment:
I'll like the way you define and autowire the TestPrintRequestRepository object but doesn't defining Autowired attributes outside IntegrationTest class result in a new spring context being created?

Reply:
Oh I hadn't considered that. I assumed it would create and wire in the bean in the current context. Why would it create a new one?

Reply:
I think currently it uses the same context for all of the int-tests. If it added a bean into the context for this test it would need to remove it for all the rest, so I think it creates a new spring-context where the context differs between tests.